nifi-nar-bundles/nifi-iceberg-bundle/nifi-iceberg-processors/src/main/java/org/apache/nifi/processors/iceberg/PutIceberg.java

```diff
@@ -299,16 +309,23 @@ public void doOnTrigger(ProcessContext context, ProcessSession session, FlowFile
             final WriteResult result = taskWriter.complete();
             appendDataFiles(context, flowFile, table, result);
         } catch (Exception e) {
-            getLogger().error("Exception occurred while writing iceberg records. Removing uncommitted data files", e);
+            final Optional<GSSException> causeOptional = findCause(e, GSSException.class, gsse -> GSSException.NO_CRED == gsse.getMajor());
+            if (causeOptional.isPresent()) {
+                getLogger().warn("No valid Kerberos credential found, retrying login", causeOptional.get());
+                initKerberosCredentials(context);
+                session.rollback(true);
```

Comment:
The HDFS PR doesn't penalize the FlowFile but does yield the processor. When these errors occur do they usually fix themselves quickly? If so I think we should go with a yield but if not, penalize is fine.

Reply:
Thanks @mattyb149 for the review. Yes you are right, yielding in case of this authentication error makes more sense than penalize. I'll update this.